src/rpc/rpc.cc

```diff
@@ -459,7 +459,7 @@ namespace rpc {
               }
               auto ptr = compress_header.get();
               auto size = read_le<uint32_t>(ptr);
-              return read_rcv_buf(in, size).then([this, size, &compressor, info] (rcv_buf compressed_data) {
+              return read_rcv_buf(in, size).then([this, size, &compressor, info, &in] (rcv_buf compressed_data) {
```

Comment:
What do we capture `&in` here for?

Reply:
The capture should be in the previous patch than because otherwise it will not compile by itself.

Reply:
Right. I first wrote the entire thing and then split it into the patches, and didn't check that each commit compiles.

Reply:
Fixed.